### ISSUE TITLE
Integrate PID Control for Movement

### ISSUE THREAD

**Issue by carrliitos** [carrliitos/cs-776-drone-swarm#10]
**Task #9 must be completed before this task.**

This task involves integrating PID control for movement (e.g., forward, backward, left, right) into the `DroneController` class to improve the drone's movement stability and responsiveness. PID control will adjust the drone's thrust based on the error between the desired and actual movement directions.

**Tasks:**
1. Research and Design: Research PID control for drone movement and design an algorithm to integrate PID control for movement into the `DroneController` class.
2. Implementation: Write the code to implement PID control for movement, using the drone's sensors to calculate the error and adjust the motor commands.
3. Testing: Test the drone's movement stability and responsiveness with PID control in the Webots simulation environment.
4. Refinement: Fine-tune the PID parameters (if necessary) to improve the drone's movement performance.

**Acceptance Criteria:**
- The drone should be able to move smoothly and accurately in the desired direction with PID control.
- PID control should improve the drone's ability to maintain stability while moving.

**Comment by carrliitos** [comment]
Closing this in favor of #12.
